modules/react/text-input/lib/InputGroup.tsx

```diff
@@ -0,0 +1,155 @@
+ * `React.cloneElement` from the [React.Children](https://react.dev/reference/react/Children) API.
+ * This means all children must be `InputGroup.*` components. Any other direct children will cause
+ * issues. You can add different elements/components inside the
+ * {@link InputGroupStart InputGroup.Start} and {@link InputGroupEnd InputGroup.End} subcomponents.
+ *
+ * ```tsx
+ * <InputGroup>
+ *   <InputGroup.Start as={SystemIcon} pointerEvents="none" icon={searchIcon} />
+ *   <InputGroup.Input />
+ *   <InputGroup.End>
+ *     <TertiaryButton tabIndex={-1} icon={xIcon} size="small" />
+ *   </InputGroup.End>
+ * </InputGroup>
+ * ```
+ */
+export const InputGroup = createComponent('div')({
+  displayName: 'InputGroup',
+  Component({children, ...elemProps}: ExtractProps<typeof Flex>, ref, Element) {
+    const isRTL = useIsRTL();
+    let paddingInlineStart: string;
+    let paddingInlineEnd: string;
+    const offsetsStart: string[] = [];
+    const offsetsEnd: string[] = [];
+
+    React.Children.forEach(children, child => {
+      if (React.isValidElement<any>(child) && child.type === InputGroupStart) {
+        const width = child.props.width || space.xl;
+        offsetsStart.push(paddingInlineStart);
+        paddingInlineStart = paddingInlineStart
+          ? `calc(${toPx(paddingInlineStart)} + ${width})`
+          : width;
+      }
+      if (React.isValidElement<any>(child) && child.type === InputGroupEnd) {
+        const width = child.props.width || space.xl;
+        offsetsEnd.push(paddingInlineEnd);
+        paddingInlineEnd = paddingInlineEnd ? `calc(${toPx(paddingInlineEnd)} + ${width})` : width;
+      }
+    });
+
+    // reverse the end to ensure the elements appear in the right order
+    offsetsEnd.reverse();
+
+    let startIndex = 0;
+    let endIndex = 0;
+
+    const mappedChildren = React.Children.map(children, child => {
+      if (React.isValidElement<any>(child)) {
+        if (child.type === InputGroupInput) {
+          return React.cloneElement(child, {paddingInlineStart, paddingInlineEnd});
+        }
+        if (child.type === InputGroupStart) {
+          const offset = offsetsStart[startIndex] || 0;
+          startIndex++;
+
+          return React.cloneElement(child, {
+            left: isRTL ? undefined : offset,
+            right: isRTL ? offset : undefined,
+          });
+        }
+        if (child.type === InputGroupEnd) {
+          const offset = offsetsEnd[endIndex] || 0;
+          endIndex++;
+
+          return React.cloneElement(child, {
+            left: isRTL ? offset : undefined,
+            right: isRTL ? undefined : offset,
+          });
+        }
+      }
+      return child;
+    });
+
+    return (
+      <Flex ref={ref} as={Element} position="relative" {...elemProps}>
+        {mappedChildren}
+      </Flex>
+    );
+  },
+  subComponents: {
+    /**
+     * A component to show at the start of the input. The input's padding will be adjusted to not
```

Comment:
not sure if we want to mention the word "inside" here. Cause start can be outside.

Reply:
I updated